### DISCUSSION TITLE
RPI Zero + Multiple USB Ethernet dongles + HTTPS

### DISCUSSION THREAD

**Discussion by MoronGamer** [rsta2/circle#297] [Q&A] [answered]
Hi there,

  I have a project which I need to communicate with an ethernet RFID reader, an ethernet ADAM I/O and a backend web API over HTTPS.  I'm thinking of connecting the RFID reader and the ADAM I/O directly using crossover cables so there's no need to use a network hub.

  Questions:

  Is it possible for a RPI Zero to communicate with 3 USB Ethernet dongles over an external usb hub using Circle?  If so, does the hub/dongle need to be specific brand/chipset?  And what about RPI 3/4 using on board ethernet + 2 USB ethernet dongles?
  
  As for communicating with a server over https, is there any supported library that I can use?

Thank you!
Alfred

**Comment by hpingel** [comment]
HTTPS you can get via mbedtls which is part of circle-stdlib. But AFAIK there are no drivers in circle for USB-ethernet dongles or rfid readers.

**Comment by dlbogdan** [comment]
Well, being that all those devices are ethernet, I assume with proper documentation one can write drivers for them. 
as for the driver for a USB Ethernet adapter, circle must have at least one driver. The embedded usb to ethernet chipset. Maybe if you can manage to get that as a dongle things would work fine.

**Reply by dlbogdan** [reply]
This MIGHT work:
https://ro.mouser.com/new/microchip/microchip-evb-lan7800lc-1-dongle/

**Comment by rsta2** [comment, marked as answer]
Unfortunately this is not possible, because Circle can handle only one active network interface at a time. Yes, there aren't drivers for specific USB Ethernet NICs in Circle, only a driver for the QEMU CDC Ethernet NIC. TLS support is in the [circle-stdlib project](https://github.com/smuehlst/circle-stdlib), which includes Circle as a submodule.

**Reply by MoronGamer** [reply]
Thank you!
